frontend/src/js/query-node-editor/useAutoLabel.ts

```diff
@@ -31,12 +31,16 @@ export function useAutoLabel({ node, onUpdateLabel }: AutoLabelProps) {
       formatConceptLabels(node.label.split("_")) === autoLabel,
   );
 
-  // save auto generated node label
-  useEffect(() => {
-    if (autoLabelEnabled && autoLabel && node.label !== autoLabel) {
-      onUpdateLabel(autoLabel);
-    }
-  }, [autoLabel, autoLabelEnabled, node.label, onUpdateLabel]);
+  const onUpdateLabelRef = useRef(onUpdateLabel);
+  onUpdateLabelRef.current = onUpdateLabel;
```

Comment:
This fixes the infinite loop. `onUpdateLabel` isn't memoized for `FormQueryNodeEditor`, so a new function reference caused this useEffect to re-run infinitely. Unfortunately it's hard to memoize the function inside `FormQueryNodeEditor` because it depends on the entire concept field value (unlike in `StandardQueryNodeEditor`). So using a ref here allows us to
- use the latest reference of the `onUpdateLabel` function
- only trigger the useEffect to run when one of `node.label` / `autoLabel` / `autoLabelEnabled` changes